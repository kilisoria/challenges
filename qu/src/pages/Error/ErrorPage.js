import React from "react";

import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import { Alert, AlertTitle } from '@material-ui/lab';
import Button from '@material-ui/core/Button';

const ErrorPage = () => {
  return (
    <Container maxWidth="md">
      <Typography component="div" style={{ backgroundColor: '#A94951', paddingTop: 10, height: '80vh' }}>
        <Alert severity="error">
          <AlertTitle>Oh, no! We ran into an issue.</AlertTitle>
  This is embarassing.<strong>We can't find what you were looking for.</strong>
        </Alert>
        <div style={{ textAlign: 'center', marginTop: 40 }}>
          <Button variant="contained" color="primary" href="/">
            Back
         </Button>
        </div>
      </Typography>
    </Container>
  );
};

export default ErrorPage;
