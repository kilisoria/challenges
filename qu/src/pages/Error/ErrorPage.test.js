import React from 'react';
import renderer from 'react-test-renderer';

import ErrorPage from './ErrorPage';

describe('GIVEN ErrorPage component', () => {
  describe('WHEN renders by default', () => {
    it('THEN should match the snapshot', () => {
      const tree = renderer.create(<ErrorPage/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
