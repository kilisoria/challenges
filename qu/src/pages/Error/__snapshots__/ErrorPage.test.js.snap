// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN ErrorPage component WHEN renders by default THEN should match the snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd"
>
  <div
    className="MuiTypography-root MuiTypography-body1"
    style={
      Object {
        "backgroundColor": "#A94951",
        "height": "80vh",
        "paddingTop": 10,
      }
    }
  >
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        <div
          className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Oh, no! We ran into an issue.
        </div>
        This is embarassing.
        <strong>
          We can't find what you were looking for.
        </strong>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": 40,
          "textAlign": "center",
        }
      }
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        href="/"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Back
        </span>
      </a>
    </div>
  </div>
</div>
`;
